### PR TITLE
Transform visibility when height/width is set.

### PR DESCRIPTION
**Pull request by sainthkh**
- Partially solves #5974

### User facing changelog

When an element uses both `transform` and `height/width`, its bounding client rect becomes 0, but the child elements are visible. This visibility bug is fixed. 

#### Why was this change necessary?

Visibility false negative

#### What is affected by this change?

`elIsHiddenByTransform` became a bit complicated, because it doesn't use `getBoundingClientRect` but a bit complicated 3D math.

#### Any implementation details to explain?

Check links in the comment. 

### How has the user experience changed?

Before: false positive. 
After: Now, it passes. 

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [NA] Have API changes been updated in the [`type definitions`](cli/types/index.d.ts)?
- [NA] Have new configuration options been added to the [`cypress.schema.json`](cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by sainthkh** [comment]
Luckily, I got the number 6000.

**Comment by sainthkh** [comment]
Another failure because of Page Load event.

File: packages/driver/src/dom/visibility.js [hunk 1]
```diff
@@ -185,19 +195,28 @@ const elIsHiddenByTransform = ($el) => {
   if (transform.startsWith('matrix3d')) {
     const m3d = transform.substring(8).match(numberRegex)
 
-    // Z Axis values
-    if (+m3d[2] === 0 && +m3d[6] === 0 && +m3d[10] === 0) {
+    // Test scale to 0
+    if ((+m3d[0] === 0 && +m3d[4] === 0 && +m3d[8] === 0) || // X Axis
+    (+m3d[1] === 0 && +m3d[5] === 0 && +m3d[9] === 0) || // Y Axis
+    (+m3d[2] === 0 && +m3d[6] === 0 && +m3d[10] === 0)) { // Z Axis
```
Comment (jennifer-shehane) [inline packages/driver/src/dom/visibility.js]: Can you pull this out into separate functions that explain what they mean? It's hard to read what this if statement is checking exactly just reading all the matrix values together.

For example, something like

```js
const xAxisIs0 = +m3d[0] === 0 && +m3d[4] === 0 && +m3d[8] === 0


// later

if (xAxisIs0 || ...)
```

File: packages/driver/src/dom/visibility.js [hunk 2]
```diff
@@ -185,19 +195,28 @@ const elIsHiddenByTransform = ($el) => {
...
     }
+
+    // Test rotate 90deg
+    const defaultNormal = [0, 0, -1]
+    const elNormal = findNormal(m3d)
+    // Simplified dot product.
+    // [0] and [1] are always 0
+    const dot = defaultNormal[2] * elNormal[2]
+
+    return Math.abs(dot) <= 1e-10
```
Comment (jennifer-shehane) [inline packages/driver/src/dom/visibility.js]: Maybe the same with this, pull this out into a function. `isRotated90Deg` or whatever this is checking.

File: packages/driver/src/dom/visibility.js [hunk 3]
```diff
@@ -185,19 +195,28 @@ const elIsHiddenByTransform = ($el) => {
...
-  if (transform !== 'none') {
-    const { width, height } = el.getBoundingClientRect()
+  const m = transform.match(numberRegex)
 
-    if (width === 0 || height === 0) {
-      return true
-    }
+  if ((+m[0] === 0 && +m[2] === 0) || // X Axis
+    (+m[1] === 0 && +m[3] === 0)) { // Y Axis
+    return true
```
Comment (jennifer-shehane) [inline packages/driver/src/dom/visibility.js]: Same - pull out into functions with descriptive variable names.

**Comment by sainthkh** [comment]
@jennifer-shehane Refactor finished.

**Comment by Bkucera** [comment]
this caused an example recipe to fail https://circleci.com/gh/cypress-io/cypress-test-example-repos/71863

**Comment by sainthkh** [comment]
@Bkucera I opened an issue in #6093. It'll be fixed in #5916.

**Pull request merged**